File: C++/Algoritmos_Orden/Bubble.cpp
#include <iostream>

using namespace std;

void bubblesort(int arreglo[], int largo)
{
    for (int indice = 0; indice < largo; indice)
    {
        bool Swap = false;

        for(int j = 0; j < largo - indice - 1; j++) //Los ultimos indices estarán acomodados
        {
            if (arreglo[j] > arreglo[j +1])         //Pregunta si el indice actual es mayor que el siguiente
            {
                int elegido = arreglo[j];
                arreglo[j] = arreglo[j +1];
                arreglo[j +1] = elegido;
                Swap = true;
            }
        }
        if (!Swap)
        {
            break;
        }
    }
}

int main()
{
    int arreglo[] = {15,16,11,13,14};
    int largo = sizeof(arreglo) / sizeof(arreglo[0]);

    cout << "Antes de ordenar los elementos del array son: ";
    for (int indice = 0; indice < largo; indice++)
    {
        cout << arreglo[indice] << " ";
    }

    bubblesort(arreglo, largo);

    cout << "Después de ordenar los elementos del array son: ";
    for (int indice = 0; indice < largo; indice++) {
        cout << arreglo[indice] << " ";
    }
    return 0;
}

File: JavaScript/Algoritmos_Orden/Bubble.js
function bubbleSort(arreglo) {
    let largo = arreglo.length;
    for (let indice = 0; indice < largo; indice++) 
    {
        let Swapped = false;
        for (let j = 0; j < largo - indice - 1; j++) 
        {
            if (arreglo[j] > arreglo[j + 1]) 
            {
                // Intercambio variables directamente
                [arreglo[j], arreglo[j + 1]] = [arreglo[j + 1], arreglo[j]];
                Swapped = true;
            }
        }
        if (Swapped === false) 
            {
            break;
        }
    }
}

let arreglo = [15, 16, 11, 13, 14];

console.log("Antes de ordenar los elementos del array son: ");
for (const j of arreglo) {
    console.log(j + " ");
}
console.log();

bubbleSort(arreglo);

console.log("Después de ordenar los elementos del array son: ");
for (let j = 0; j < arreglo.length; j++) {
    console.log(arreglo[j] + " ");
}
console.log();